sdk/test-utils/recorder-new/MIGRATION.md

```diff
@@ -0,0 +1,278 @@
+
+```ts
+await recorder.start({
+  envSetupForPlayback: {
+    // Your environment variables (equivalent to the old recorder's replaceableVariables option). See the section on environment variables below for detail
+  },
+  // Other options, e.g. sanitizers (which replace the customizationsOnRecordings option)
+});
+```
+
+And at the end of your test (or in an `afterEach` block), stop the recorder:
+
+```ts
+await recorder.stop();
+```
+
+It is important that `recorder.stop()` is called, or otherwise the next test will throw an error when trying to start the already started recorder. Additionally, it is important that both the `start` and `stop` calls are awaited, for similar reasons.
+
+## Environment variables
+
+In the legacy recorder, the `replaceableVariables` option could be used to specify environment variables that would be replaced in the recording and set during playback. This could be used to ensure that secrets and user-specific options do not appear in the recording body.
+
+The Unified Recorder client provides this functionality through the use of the `envSetupForPlayback` option, which is passed when `recorder.start` is called. Like the legacy recorder, it takes in an object mapping environment variables to what they should be replaced with in the recording. For example:
+
+```ts
+await recorder.start({
+  envSetupForPlayback: {
+    TABLES_SAS_CONNECTION_STRING: "fakeConnectionString",
+  },
+});
+```
+
+Under the hood, this is powered by the Unified Recorder's sanitizer functionality.
+
+**⚠️Important:** To access environment variables, you must use the `env` export made available from the **new** recorder. This ensures that environment variables are sourced from the correct location (using `process.env` and `dotenv` in Node, and using `window.__env__` via Mocha in the browser), and also means that the environment variables set in `envSetupForPlayback` are used in playback mode.
+
+## Recorder variables
+
+If you want to compute a value at record time and re-use it during playback, the Unified Recorder's variable functionality is for you. This API lets you declare variables which are stored with the recording at record time. During playback, instead of computing the variable afresh, the value will be retrieved from the recording. A use case of this might be to set a value randomly during record time that needs to be the same during playback.
+
+Here is an example:
+
+```ts
+const queueName = recorder.variable("queueName", "queue-${Math.floor(Math.random * 1000)}");
+// Assume that we have a client that has a createQueue method.
+await client.createQueue(queueName);
+```
+
+In this example, the name of the queue used in the recording is randomized. However, in playback, instead of using the value passed into `recorder.variable`, the value will be retrieved from the recording file. This means that the name of the queue will be consistent between recording and playback.
```

Comment:
```suggestion
In this example, the name of the queue used in the recording is randomized. However, in playback, instead of using the value passed into `recorder.variable`, the value will be retrieved from the recording file. This means that the name of the queue will be consistent between record and playback modes.
```

reads better?

Reply:
No real difference imho, but sure, if you like